third_party/tonic/converter/dart_converter.h

```diff
@@ -279,15 +281,80 @@ struct DartConverter<const char*> {
 ////////////////////////////////////////////////////////////////////////////////
 // Collections
 
+inline Dart_Handle LookupNonNullableType(const std::string& library_name,
+                                         const std::string& type_name) {
+  auto library =
+      Dart_LookupLibrary(DartConverter<std::string>::ToDart(library_name));
+  if (LogIfError(library)) {
+    return library;
+  }
+  auto type_string = DartConverter<std::string>::ToDart(type_name);
+  if (LogIfError(type_string)) {
+    return type_string;
+  }
+  auto type = Dart_GetNonNullableType(library, type_string, 0, nullptr);
+  if (LogIfError(type)) {
+    return type;
+  }
+  return type;
+}
+
+template <typename T,
+          std::enable_if_t<std::is_same<std::string, T>::value, int> = 0>
+Dart_Handle ToDartTypeHandle() {
+  return LookupNonNullableType("dart:core", "String");
+}
+
+template <typename T, std::enable_if_t<std::is_integral<T>::value, int> = 0>
+Dart_Handle ToDartTypeHandle() {
+  return LookupNonNullableType("dart:core", "int");
+}
+
+template <typename T,
+          std::enable_if_t<std::is_floating_point<T>::value, int> = 0>
+Dart_Handle ToDartTypeHandle() {
+  return LookupNonNullableType("dart:core", "double");
+}
+
+template <typename T>
+Dart_Handle CreateZeroInitializedDartObject(
+    Dart_Handle type_handle_or_null = ::Dart_Null()) {
+  if constexpr (std::is_same<std::string, T>::value) {
+    return ::Dart_EmptyString();
+  } else if constexpr (std::is_integral<T>::value) {
+    return ::Dart_NewIntegerFromUint64(0u);
+  } else if constexpr (std::is_floating_point<T>::value) {
+    return ::Dart_NewDouble(0.0);
+  } else {
+    auto object = ::Dart_New(type_handle_or_null, ::Dart_Null(), 0, nullptr);
```

Comment:
> It doesn't look like CreateZeroInitializedDartObject is ever passed a null object as the type.

It is used above in [`//fml/dart/dart_converter.h`](https://github.com/flutter/engine/pull/21820/files#diff-30826356f1934632f024ffea52352e2d42840066513868f17651fccfc1c7a4cfR28).

`type_handle_or_null` is only `null` when the type can be matched by one of the type traits. In reality, it can always be `null` because we can use `ToDartTypeHandle` here to get the handle. But I left this in as an optimization because callers always have a handle to the type (like in the case of lists). And, in that case, I didn't want to resolve the library and lookup the type in it again. I can get rid of the argument if it makes things easier to read.

Reply:
Ok, I understand it will be null for string, is_integral and is_floating_point.
Can we add an assert in the final else that type_handle_or_null is not ::Dart_Null()